src/ngx_http_lua_socket_tcp.c

```diff
@@ -3571,6 +3877,257 @@ ngx_http_lua_socket_tcp_finalize_write_part(ngx_http_request_t *r,
 }
 
 
+static void
+ngx_http_lua_socket_tcp_conn_op_timeout_handler(ngx_event_t *ev)
+{
+    ngx_http_lua_socket_tcp_upstream_t      *u;
+    ngx_http_lua_ctx_t                      *ctx;
+    ngx_connection_t                        *c;
+    ngx_http_request_t                      *r;
+    ngx_http_lua_co_ctx_t                   *coctx;
+    ngx_http_lua_loc_conf_t                 *llcf;
+    ngx_http_lua_socket_tcp_conn_op_ctx_t   *conn_op_ctx;
+
+    conn_op_ctx = ev->data;
+    ngx_queue_remove(&conn_op_ctx->queue);
+
+    u = conn_op_ctx->u;
+    r = u->request;
+
+    coctx = u->write_co_ctx;
+    coctx->cleanup = NULL;
+    /* note that we store conn_op_ctx in coctx->data instead of u */
+    coctx->data = conn_op_ctx;
+    u->write_co_ctx = NULL;
+
+    llcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_module);
+
+    if (llcf->log_socket_errors) {
+        ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
+                      "lua tcp socket queued connect timed out,"
```

Comment:
note for merge: remove comma

Reply:
@thibaultcha You can address such minor things yourself during the merge?

Reply:
Yes, will do.